processor/src/test/java/org/mapstruct/ap/test/conversion/uuid/UUIDConversionTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.conversion.uuid;
+
+import java.util.UUID;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+import org.mapstruct.ap.testutil.runner.GeneratedSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests conversions between {@link java.util.UUID} and String.
+ *
+ * @author Jason Bodnar
+ */
+@RunWith(AnnotationProcessorTestRunner.class)
+public class UUIDConversionTest {
+
+    private final GeneratedSource generatedSource = new GeneratedSource();
+
+    @Rule
+    public GeneratedSource getGeneratedSource() {
+        return generatedSource;
+    }
+
+    @Test
+    @IssueKey("2391")
+    @WithClasses({ UUIDSource.class, UUIDTarget.class, UUIDMapper.class })
+    public void shouldApplyUUIDConversion() {
+        UUIDSource source = new UUIDSource();
+        source.setUUIDA( UUID.randomUUID() );
+
+        UUIDTarget target = UUIDMapper.INSTANCE.sourceToTarget( source );
+
+        assertThat( target ).isNotNull();
+        assertThat( target.getUUIDA() ).isEqualTo( source.getUUIDA().toString() );
+    }
+
+    @Test
+    @IssueKey("2391")
+    @WithClasses({ UUIDSource.class, UUIDTarget.class, UUIDMapper.class })
+    public void shouldApplyReverseUUIDConversion() {
+        UUIDTarget target = new UUIDTarget();
+        target.setUUIDA( UUID.randomUUID().toString() );
+
+        UUIDSource source = UUIDMapper.INSTANCE.targetToSource( target );
+
+        assertThat( source ).isNotNull();
+        assertThat( source.getUUIDA() ).isEqualTo( UUID.fromString( target.getUUIDA() ) );
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
Let's use `assertThatThrownBy` from AssertJ instead of `Test#expected`

Reply:
Fixed.